app/containers/SongContainer/index.js

```diff
@@ -0,0 +1,108 @@
+/**
+ *
+ * SongContainer Container
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { injectSaga } from 'redux-injectors';
+import { selectItuneData, selectArtistName } from './selectors';
+import { Input } from 'antd';
+import ItuneCard from '@app/components/ItuneCard';
+import { songContainerCreators } from './reducer';
+import For from '@app/components/For';
+import saga from './saga';
+import get from 'lodash/get';
+import styled from 'styled-components';
+
+const StyledInput = styled(Input)`
+  && {
+    width: 25rem;
+    height: 2.5rem;
+    margin-bottom: 1rem;
+  }
+`;
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+`;
+
+const CardWrapper = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  justify-content: center;
+  gap: 1rem;
+`;
+
+export function SongContainer({ artistName, ituneData, dispatchItuneSongs, dispatchClearSongs }) {
+  const changeHandler = (evt) => {
+    let searchTerm = evt.target.value;
+    if (searchTerm) {
+      dispatchItuneSongs(searchTerm);
+    } else {
+      dispatchClearSongs();
+    }
+  };
+
```

Comment:
can we please add a debounce to the dispatch

Reply:
done